src/dsl/dynamic.jl

```diff
@@ -9,14 +10,30 @@ function arg_to_ast(arg::Argument)
     ast
 end
 
+"Escape argument defaults (if present)."
 function escape_default(arg)
     (arg.default == nothing ? nothing :
         Expr(:call, :Some, esc(something(arg.default))))
 end
 
+"Rewrites :gentrace and :genparam with their dynamic implementations."
+function rewrite_dynamic_gen_exprs(expr)
+    MacroTools.postwalk(expr) do e
```

Comment:
For my own education: is `postwalk` an arbitrary choice, or is it the only kind of walk that would work here?

Reply:
You could also use `prewalk` here, since the expression-rewriting doesn't cause any infinite loops, but `postwalk` is recommended by default --- `postwalk` walks the AST bottom-up, and so it's easier to reason about what it's doing. I've only had to use `prewalk` when walking the AST top-down was specifically required, e.g. when doing `extract_quoted_expressions`.